istio-telemetry/kiali-operator/templates/clusterrole.yaml

```diff
@@ -0,0 +1,302 @@
+{{- if .Values.kialiOperator.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kiali-operator
+  labels:
+    app: kiali-operator
+    release: {{ .Release.Name }}
+rules:
+  - apiGroups: [""]
+    resources:
+    - configmaps
+    - endpoints
+    - events
+    - persistentvolumeclaims
+    - pods
+    - serviceaccounts
+    - services
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: [""]
+    resources:
+    - namespaces
+    verbs:
+    - get
+    - list
+    - patch
+  - apiGroups: ["apps"]
+    resources:
+    - deployments
+    - replicasets
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["monitoring.coreos.com"]
+    resources:
+    - servicemonitors
+    verbs:
+    - create
+    - get
+  - apiGroups: ["apps"]
+    resourceNames:
+    - kiali-operator
+    resources:
+    - deployments/finalizers
+    verbs:
+    - update
+  - apiGroups: ["kiali.io"]
+    resources:
+    - '*'
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["rbac.authorization.k8s.io"]
+    resources:
+    - clusterrolebindings
+    - clusterroles
+    - rolebindings
+    - roles
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources:
+    - customresourcedefinitions
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups: ["extensions"]
+    resources:
+    - ingresses
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["route.openshift.io"]
+    resources:
+    - routes
+    verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups: ["oauth.openshift.io"]
```

Comment:
assume this has been tested on pristine K8s and works correctly without some of these apiGroups?

Reply:
Yes, this all works on pristine k8s (we test on minikube, for example). k9s allows roles to be given access to undefined apiGroups.